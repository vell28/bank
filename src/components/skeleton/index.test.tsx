import React from 'react';

import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Skeleton } from '.';

const SkeletonRender = () => <div className="skeleton">Any component</div>;

describe('<Skeleton /> tests', () => {
  test('should success render, should render content if not loading', () => {
    const component: any = mount(
      <Skeleton render={<SkeletonRender />} isLoading={false}>
        <div className="any-content">Any</div>
      </Skeleton>,
    );

    const skeleton = component.find('.skeleton');
    expect(skeleton.length).toEqual(0);

    const content = component.find('.any-content');

    expect(content.length).toEqual(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should success render, should render skeleton if loading', () => {
    const component: any = mount(
      <Skeleton render={<SkeletonRender />} isLoading>
        <div className="any-content">Any</div>
      </Skeleton>,
    );

    const skeleton = component.find('.skeleton');
    expect(skeleton.length).toEqual(1);

    const content = component.find('.any-content');

    expect(content.length).toEqual(0);
    expect(toJson(component)).toMatchSnapshot();
  });
});
