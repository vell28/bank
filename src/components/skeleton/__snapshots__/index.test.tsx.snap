// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> tests should success render, should render content if not loading 1`] = `
<Skeleton
  isLoading={false}
  render={<SkeletonRender />}
>
  <div
    className="any-content"
  >
    Any
  </div>
</Skeleton>
`;

exports[`<Skeleton /> tests should success render, should render skeleton if loading 1`] = `
<Skeleton
  isLoading={true}
  render={<SkeletonRender />}
>
  <SkeletonRender>
    <div
      className="skeleton"
    >
      Any component
    </div>
  </SkeletonRender>
</Skeleton>
`;
